CONTRIBUTING.md

```diff
@@ -0,0 +1,518 @@
+
+   ```rust
+   // Parse the Excel xls Shared String Table (SST). See [MS-XLS] 2.4.265.
+   //
+   // https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-xls/b6231b92-d32e-4626-badd-c3310a672bab
+   fn parse_sst() -> Result<(), XlsError>
+   ```
+
+## Linting with Clippy
+
+We use Clippy to catch common mistakes and improve code quality:
+
+```bash
+# Run clippy.
+cargo clippy --lib --examples --tests
+```
+
+Note `cargo clippy --all-features --all-targets` only works with `+nightly` due
+to the `benches` benchmark file which requires it.
+
+You can also run Clippy with specific feature sets:
+
+```bash
+cargo clippy --features "chrono"
+cargo clippy --no-default-features
+```
+
+## Testing
+
+### Test Categories
+
+1. **Unit Tests**: Test individual functions or methods.
+2. **Integration Tests**: Test parsing real spreadsheet files.
+3. **Documentation Tests**: Ensure the examples in docs work.
+
+### Running Tests
+
+The different test variants can be run as follows:
+
+```bash
+# Run all tests.
+cargo test
+
+# Run tests with all features.
+cargo test --all-features
+
+# Run integration tests only.
+cargo test --test '*'
+
+# Run lib tests only.
+cargo test --lib
+
+# Run specific test.
+cargo test test_name
+
+# Run tests in a specific module.
+cargo test xlsx::
+```
+
+### Writing Tests
+
+1. **Unit tests** are generally in the same file as the code being tested:
+   ```rust
+   #[cfg(test)]
+   mod tests {
+       use super::*;
+
+       #[test]
+       fn test_parse_number() {
+           let data = &[0x01, 0x02, 0x03, 0x04];
+           let result = parse_number(data);
+           assert_eq!(result, 67305985);
+       }
+   }
+   ```
+
+   Unit tests may move to separate files as their size increases.
+
+2. **Integration tests** should be in the `tests/test.rs` file:
+
+   ```rust
+   #[test]
+   // Comment explaining what is being tested and the `#123` GitHub
+   // issue number if available.
+   fn test_open_sample_file() {
+       let mut workbook: Xlsx<_> = wb("issues.xlsx");
+       // Test specific functionality.
+   }
+   ```
+
+3. **Test file naming**: Use descriptive names like
+   `tests/no_shared_string_table.xlsx` for issue-specific test files.
+
+### Test Data
+
+- Add test files to the `tests/` directory.
+- Use minimal files that reproduce specific issues.
+- Document in the comments what each test file is testing and refer to the
+  GitHub issue number, if applicable, see the example above.
+
+## Performance Testing
+
+### Benchmarking
+
+The `benches` benchmark test can be used to check for any major performance
+regressions. Running the benchmarking requires `+nightly`:
+
+```bash
+# Run benchmarks.
+cargo +nightly bench
+```
+
+### Performance Guidelines
+
+1. **Avoid unnecessary allocations** in hot paths.
+2. **Consider the memory usage** when processing large files. Users of
+   `calamine` tend to process large files.
+3. **Profile before optimizing** - use tools like `perf` or `flamegraph`.
```

Comment:
https://github.com/mstange/samply is really nice and opens up the flamegraph in the firefox profiler which lets you press a button to upload and share them.